src/wallet/wallet.cpp

```diff
@@ -1768,6 +1769,15 @@ CWallet::ScanResult CWallet::ScanForWalletTransactions(const uint256& start_bloc
             // scan succeeded, record block as most recent successfully scanned
             result.last_scanned_block = block_hash;
             result.last_scanned_height = block_height;
+
+            if (save_progress && next_interval) {
+                CBlockLocator loc= m_chain->getActiveChainLocator(block_hash);
+
+                if (!loc.IsNull()) {
+                    WalletBatch batch(GetDatabase());
+                    batch.WriteBestBlock(loc);
```

Comment:
nit:
We probably could log this for debugging purposes:
```WalletLogPrintf("Saving scan progress %d.\n", block_height);```

As it's done once every 60 seconds, it will not spam the log much.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/25036/commits/a89ddfbe22b6db5beda678c9493e08fec6144122. Thanks